src/vm.rs

```diff
@@ -172,30 +255,30 @@ impl<VCpuType: VirtualCPU> UhyveVm<VCpuType> {
 			unsafe { self.mem.as_slice_mut() } // slice only lives during this fn call
 				.try_into()
 				.expect("Guest memory is not large enough for pagetables"),
+				self.mem.guest_address.as_u64()
 		);
 	}
 
 	pub fn load_kernel(&mut self) -> LoadKernelResult<()> {
+		// TODO: Remove the duplicate load in load_kernel.
```

Comment:
This part is presumably / hopefully what causes the current revision (https://github.com/hermit-os/uhyve/pull/725/commits/f806c0f8c7db2cf2a054f5c97732cf6aabc05865) to break.

I experienced a massive problem here, as I could not create a new entry for `elf` or `object` (as we initially read `object` in `new` and `load_kernel` has to see it somehow...) nor create a singleton such as `OnceLock` that would allow me to access the object from `load_kernel` anyways.

The current version returns the following:
```
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.20s
     Running `target/debug/uhyve -v data/x86_64/rusty_demo`
[2024-07-22T20:19:58Z INFO  hermit_entry::elf] Parsing kernel from ELF at 0x7f0203916010..0x7f0203a3cc50 (len = 0x126c40 B / 1207360 B)
[2024-07-22T20:19:58Z DEBUG uhyvelib::linux::x86_64::kvm_cpu] Initialize interrupt controller
[2024-07-22T20:19:58Z DEBUG uhyvelib::vm] Initialize guest memory
[2024-07-22T20:19:58Z INFO  hermit_entry::elf] Parsing kernel from ELF at 0x560e87f85020..0x560e880abc60 (len = 0x126c40 B / 1207360 B)
thread 'main' panicked at src/vm.rs:269:28:
attempt to subtract with overflow
stack backtrace:
   0: rust_begin_unwind
             at /rustc/5affbb17153bc69a9d5d8d2faa4e399a014a211e/library/std/src/panicking.rs:665:5
   1: core::panicking::panic_fmt
             at /rustc/5affbb17153bc69a9d5d8d2faa4e399a014a211e/library/core/src/panicking.rs:74:14
   2: core::panicking::panic_const::panic_const_sub_overflow
             at /rustc/5affbb17153bc69a9d5d8d2faa4e399a014a211e/library/core/src/panicking.rs:181:21
   3: uhyvelib::vm::UhyveVm<VCpuType>::load_kernel
             at ./src/vm.rs:269:22
   4: uhyvelib::linux::<impl uhyvelib::vm::UhyveVm>::run
             at ./src/linux/mod.rs:80:3
   5: uhyve::run_uhyve
             at ./src/bin/uhyve.rs:287:2
   6: uhyve::main
             at ./src/bin/uhyve.rs:291:16
   7: core::ops::function::FnOnce::call_once
             at /rustc/5affbb17153bc69a9d5d8d2faa4e399a014a211e/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

This works better if we force `let start_address = object.start_addr().unwrap_or(0x40000);` against hello_c, which does not make use of ASLR.

Reply:
https://github.com/n0toose/uhyve/commit/946c91bf428543320983a0c3ca499074076ae715

I made the exact same mistake with the offset here lol